web-ui/src/app/components/block-details/block-details.component.ts

```diff
@@ -26,11 +34,24 @@ export class BlockDetailsComponent implements OnInit {
     private errorService: ErrorService) { }
 
   ngOnInit() {
-    this.route.params.forEach(params => this.onBlockhash(params['query']));
+    this.route.params.forEach((params) => {
```

Comment:
the idea for the `onBlockhash` function is to react to a new blockhash event, please move the logic there.